Source/WebCore/editing/DeleteSelectionCommand.cpp

```diff
@@ -420,6 +426,9 @@ void DeleteSelectionCommand::saveTypingStyleState()
         return;
     }
 
+    if (shouldNotInheritStyleFrom(*m_selectionToDelete.start().anchorNode()))
```

Comment:
I understand the Chromium project already made this change, but I think it might not be entirely correct.

Makes sense that we would not want to inherit from, say, an image element but wouldn’t we want to inherit from what’s before/after that element? Like if the text after the image that we were deleting was bold? Test case might be something like either of these:

    <b>This is bold <img></b> and this is not
    <b>This is bold <img> yes, bold</b> and this is not

Deleting a range starting before the image and ending after the word "and". We’d want the typing style to be bold in both cases, I think. Maybe this isn’t the right test case. I’d like to see a lot more test cases to be sure we got this right.

Reply:
Right, this is precisely why I stated Blink fix is wrong in https://bugs.webkit.org/show_bug.cgi?id=119116